mlflow/bedrock/_autolog.py

```diff
@@ -119,6 +123,36 @@ def _patched_converse(original, self, *args, **kwargs):
     ) as span:
         # NB: Bedrock client doesn't accept any positional arguments
         span.set_inputs(kwargs)
-        result = original(self, *args, **kwargs)
-        span.set_outputs(result)
+
+        if tool_config := kwargs.get("toolConfig"):
+            try:
+                tools = [convert_tool_to_mlflow_chat_tool(tool) for tool in tool_config["tools"]]
+                set_span_chat_tools(span, tools)
+            except Exception as e:
+                _logger.debug(f"Failed to set tools for {span}. Error: {e}")
+
+        result = None
+        try:
+            result = original(self, *args, **kwargs)
+            span.set_outputs(result)
+        finally:
+            _set_chat_messages_attributes(span, kwargs, result)
         return result
+
+
+def _set_chat_messages_attributes(span, kwargs, response: Optional[dict[str, Any]]):
```

Comment:
nit: since only "messages" are used from kwargs, we could receive messages instead of kwargs as a whole.